drivers/claims_reporting/spec/reconciliation_report_spec.rb

```diff
@@ -0,0 +1,222 @@
+             pending_disenrollment_date: month.end_of_month,
+             disenrollment_date: nil)
+      create(:careplan, patient: patient)
+      create(:qualifying_activity,
+             patient: patient,
+             date_of_activity: month.beginning_of_month + 11.days,
+             claim_submitted_on: claim_submitted_on)
+      active_patients << [patient, 0, 1, []]
+    end
+    # 4. after grace period, before pending dis-enrollment, after careplan signed by engagement period end
+    create(:patient).tap do |patient|
+      create(:prior_referral,
+             patient: patient,
+             enrollment_start_date: month.beginning_of_month - (engagement_days - 10.days),
+             pending_disenrollment_date: month.end_of_month,
+             disenrollment_date: nil)
+      cp = create(:careplan, patient: patient, provider_signed_on: month.beginning_of_month)
+      create(:qualifying_activity,
+             patient: patient,
+             date_of_activity: month.beginning_of_month + 11.days,
+             claim_submitted_on: claim_submitted_on)
+      active_patients << [patient, 0, 0, [cp.provider_signed_on.to_date]]
+    end
+    # 5. after grace period, before pending dis-enrollment, after careplan signed after engagement period
+    create(:patient).tap do |patient|
+      create(:prior_referral,
+             patient: patient,
+             enrollment_start_date: month.beginning_of_month - (engagement_days - 10.days),
+             pending_disenrollment_date: month.end_of_month,
+             disenrollment_date: nil)
+      cp = create(:careplan, patient: patient, provider_signed_on: month.beginning_of_month + 11.days)
+      create(:qualifying_activity,
+             patient: patient,
+             date_of_activity: month.beginning_of_month + 15.days,
+             claim_submitted_on: claim_submitted_on)
+      active_patients << [patient, 0, 0, [cp.provider_signed_on.to_date]]
+    end
+    # 6. after grace period, before pending dis-enrollment, before careplan signed after engagement period
+    # this is ok because if a careplan is signed it covers other activities in the plan (but not itself)
+    create(:patient).tap do |patient|
+      create(:prior_referral,
+             patient: patient,
+             enrollment_start_date: month.beginning_of_month - (engagement_days - 10.days),
+             pending_disenrollment_date: month.end_of_month,
+             disenrollment_date: nil)
+      cp = create(:careplan, patient: patient, provider_signed_on: month.beginning_of_month + 15.days)
+      create(:qualifying_activity,
+             patient: patient,
+             date_of_activity: month.beginning_of_month + 11.days,
+             claim_submitted_on: claim_submitted_on)
+      active_patients << [patient, 0, 0, [cp.provider_signed_on.to_date]]
+    end
+    # 7. after pending disenrollment before actual disenrollment, after valid careplan
```

Comment:
This points to a possible issue in the current codebase... I think we should be able to enforce that if there is a `disenrollment_date`, the 'pending_disenrollment_date` is cleared...

Reply:
Agreed but there is not current validation so wanted to ensure that we did the right thing with messy data here. A separate PR to add the validation and clean up existing data perhaps?